pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/AbstractJavaRule.java

```diff
@@ -96,4 +106,54 @@ public final boolean dependsOn(AstProcessingStage<?> stage) {
         return ((JavaProcessingStage) stage).ruleDependsOnThisStage(this);
     }
 
+    public Object visit(ASTAnnotation node, Object data) {
```

Comment:
Because ASTAnnotation is `#void`in the grammar, so JJTree doesn't generate a visit method. I've been working on an alternative to JJTree that would avoid that, and cleanup our grammar and our `alljavacc.xml`s, but it's not completely ready

Reply:
But yes AbstractJavaRule is not totally in sync with JavaParserVisitorAdapter for now. I'll add a FIXME